source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -877,8 +889,19 @@ void Filter::onUpstreamEvent(Network::ConnectionEvent event) {
         }
       }
       if (!downstream_closed_) {
-        route_ = pickRoute();
-        establishUpstreamConnection();
+        if (connect_attempts_ >= config_->maxConnectAttempts()) {
```

Comment:
and if feature flag false.  the goal is to have 0 behavioral changes if the flag is false, desired future behavior when it's true